contracts/multisigs/SafeMultisigWithRecoveryModule.sol

```diff
@@ -0,0 +1,107 @@
+/// @author Aleksandr Kuperman - <aleksandr.kuperman@valory.xyz>
+/// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
+/// @author Mariapia Moscatiello - <mariapia.moscatiello@valory.xyz>
+contract SafeMultisigWithRecoveryModule {
+    // Selector of the Safe setup function
+    bytes4 public constant SAFE_SETUP_SELECTOR = 0xb63e800d;
+    // Encoded selector of the Recovery module enableModule function
+    bytes4 public constant ENABLE_MODULE_SELECTOR = 0x24292962;
+    // Default data size for several Safe Factory params without payload: address + uint256 = 2 full slots = 64 (bytes)
+    uint256 public constant DEFAULT_DATA_LENGTH = 64;
+
+    // Safe contract address
+    address public immutable safe;
+    // Safe Factory contract address
+    address public immutable safeProxyFactory;
+    // Recovery module address
+    address public immutable recoveryModule;
+
+    // Reentrancy lock
+    uint256 internal _locked = 1;
+
+    /// @dev SafeMultisigWithRecoveryModule constructor.
+    /// @param _safe Safe contract address.
+    /// @param _safeProxyFactory Safe proxy factory contract address.
+    /// @param _recoveryModule Recovery module address.
+    constructor (address _safe, address _safeProxyFactory, address _recoveryModule) {
+        // Check for zero addresses
+        if (_safe == address(0) || _safeProxyFactory == address(0) || _recoveryModule == address(0)) {
+            revert ZeroAddress();
+        }
+
+        safe = _safe;
+        safeProxyFactory = _safeProxyFactory;
+        recoveryModule = _recoveryModule;
+    }
+
+    /// @dev Creates a Safe multisig.
+    /// @param owners Set of multisig owners.
+    /// @param threshold Number of required confirmations for a multisig transaction.
+    /// @param data Encoded data related to the creation of a chosen multisig.
+    /// @return multisig Address of a created multisig.
+    function create(
+        address[] memory owners,
+        uint256 threshold,
+        bytes memory data
+    ) external returns (address multisig) {
+        // Reentrancy guard
+        if (_locked > 1) {
+            revert ReentrancyGuard();
+        }
+        _locked = 2;
+
+        address fallbackHandler;
+        uint256 nonce;
+
+        uint256 dataLength = data.length;
+        if (dataLength > 0) {
+            // Check for correct data length
+            if (dataLength != DEFAULT_DATA_LENGTH) {
+                revert IncorrectDataLength(DEFAULT_DATA_LENGTH, data.length);
+            }
+
+            // Decode fallback handler and nonce
+            (fallbackHandler, nonce) = abi.decode(data, (address, uint256));
+        }
+
+        // Convert enableModule selector into bytes
+        bytes memory payload = bytes.concat(ENABLE_MODULE_SELECTOR);
+
+        // Encode the gnosis setup function parameters
+        bytes memory safeParams = abi.encodeWithSelector(SAFE_SETUP_SELECTOR, owners, threshold, recoveryModule,
```

Comment:
so `recoveryModule` replaces `to` here https://github.com/valory-xyz/autonolas-registries/blob/main/contracts/multisigs/GnosisSafeMultisig.sol#L103C13-L103C15